pkg/broker/receiver.go

```diff
@@ -179,24 +181,40 @@ func (r *Receiver) sendEvent(ctx context.Context, tctx cloudevents.HTTPTransport
 		return nil, err
 	}
 
+	// Set up the metrics context
+	ctx, _ = tag.New(ctx,
+		tag.Insert(TagTrigger, trigger.String()),
+		tag.Insert(TagBroker, fmt.Sprintf("%s/%s", trigger.Namespace, t.Spec.Broker)),
+	)
+	// Record event count and filtering time
+	startTS := time.Now()
+	defer func() {
+		dispatchTimeMS := int64(time.Now().Sub(startTS) / time.Millisecond)
+		stats.Record(ctx, MeasureTriggerDispatchTime.M(dispatchTimeMS))
+		stats.Record(ctx, MeasureTriggerEventsTotal.M(1))
+	}()
+
 	subscriberURIString := t.Status.SubscriberURI
 	if subscriberURIString == "" {
-		r.logger.Error("Unable to read subscriberURI")
+		ctx, _ = tag.New(ctx, tag.Upsert(TagResult, "error"))
 		return nil, errors.New("unable to read subscriberURI")
 	}
 	// We could just send the request to this URI regardless, but let's just check to see if it well
 	// formed first, that way we can generate better error message if it isn't.
 	subscriberURI, err := url.Parse(subscriberURIString)
 	if err != nil {
 		r.logger.Error("Unable to parse subscriberURI", zap.Error(err), zap.String("subscriberURIString", subscriberURIString))
+		ctx, _ = tag.New(ctx, tag.Upsert(TagResult, "error"))
 		return nil, err
 	}
 
-	if !r.shouldSendMessage(&t.Spec, event) {
+	if !r.shouldSendMessage(ctx, &t.Spec, event) {
 		r.logger.Debug("Message did not pass filter", zap.Any("triggerRef", trigger))
+		ctx, _ = tag.New(ctx, tag.Upsert(TagResult, "reject"))
 		return nil, nil
 	}
 
+	ctx, _ = tag.New(ctx, tag.Upsert(TagResult, "accept"))
 	sendingCTX := SendingContext(ctx, tctx, subscriberURI)
 	return r.ceClient.Send(sendingCTX, *event)
```

Comment:
Does this function need/want to record the success/failure from the `r.ceClient.Send` call?

Reply:
Good catch! I changed this so the `result` tag has value `error` if `Send` returns an error.